### PR TITLE
doc: Sync classref with current source

### PR DESCRIPTION
**Pull request by akien-mga**
Includes various changes triggered by the refactoring of method bindings in #42780.

---

**Draft** as some of the changes might be bugs that need to be addressed before we make them harder to spot.

### CODE REVIEW

File: doc/classes/Basis.xml [hunk 1]
```diff
@@ -187,25 +187,6 @@
...
-		<method name="xform_inv">
-			<return type="Vector3">
-			</return>
-			<argument index="0" name="v" type="Vector3">
-			</argument>
-			<description>
-				Returns a vector transformed (multiplied) by the transposed basis matrix.
-				[b]Note:[/b] This results in a multiplication by the inverse of the matrix only if it represents a rotation-reflection.
-			</description>
-		</method>
```
Comment (akien-mga) [inline doc/classes/Basis.xml:208]: `Basis`, `Transform` and `Transform2D` all lost their `xform` and `xform_inv` methods. Bug?

File: doc/classes/Color.xml [hunk 2]
```diff
@@ -195,29 +195,6 @@
 				[/codeblocks]
 			</description>
 		</method>
-		<method name="from_hsv">
```
Comment (akien-mga) [inline doc/classes/Color.xml:198]: Missing binding?

File: doc/classes/Geometry2D.xml [hunk 3]
```diff
@@ -200,7 +200,8 @@
 				Inflates or deflates [code]polygon[/code] by [code]delta[/code] units (pixels). If [code]delta[/code] is positive, makes the polygon grow outward. If [code]delta[/code] is negative, shrinks the polygon inward. Returns an array of polygons because inflating/deflating may result in multiple discrete polygons. Returns an empty array if [code]delta[/code] is negative and the absolute value of it approximately exceeds the minimum bounding rectangle dimensions of the polygon.
 				Each polygon's vertices will be rounded as determined by [code]join_type[/code], see [enum PolyJoinType].
 				The operation may result in an outer polygon (boundary) and inner polygon (hole) produced which could be distinguished by calling [method is_polygon_clockwise].
-				[b]Note:[/b] To translate the polygon's vertices specifically, use the [method Transform2D.xform] method:
+				[b]FIXME:[/b] MethodBind refactoring made Transform2D.xform disappear, so this is no longer valid.
```
Comment (akien-mga) [inline doc/classes/Geometry2D.xml:203]: Needs to be fixed before merging of course.

File: doc/classes/PackedByteArray.xml [hunk 4]
```diff
@@ -20,9 +20,9 @@
 			</description>
 		</method>
 		<method name="append">
-			<return type="void">
+			<return type="bool">
```
Comment (akien-mga) [inline doc/classes/PackedByteArray.xml:23]: How come `append` and `push_back` now return `bool`? It seems to match what is defined in `core/vector.h` but it returns `true` on failure and `false` on error, is that a weird Unix-style error handling?

File: doc/classes/PackedColorArray.xml [hunk 5]
```diff
@@ -116,11 +116,20 @@
 				Returns the size of the array.
 			</description>
 		</method>
-		<method name="sort">
```
Comment (akien-mga) [inline doc/classes/PackedColorArray.xml:119]: What happened to `sort` for the `Color`, `Vector2` and `Vector3` arrays?

File: doc/classes/Quat.xml [hunk 6]
```diff
@@ -132,26 +132,6 @@
...
-		</method>
-		<method name="set_euler">
-			<return type="void">
-			</return>
-			<argument index="0" name="euler" type="Vector3">
-			</argument>
-			<description>
-				Sets the quaternion to a rotation specified by Euler angles (in the YXZ convention: when decomposing, first Z, then X, and Y last), given in the vector format as (X angle, Y angle, Z angle).
-			</description>
-		</method>
```
Comment (akien-mga) [inline doc/classes/Quat.xml:154]: Missing bindings, or removed on purpose?

File: doc/classes/Rect2i.xml [hunk 7]
```diff
@@ -133,7 +133,7 @@
 		<method name="has_point">
 			<return type="bool">
 			</return>
-			<argument index="0" name="point" type="Vector2i">
+			<argument index="0" name="point" type="Vector2">
```
Comment (akien-mga) [inline doc/classes/Rect2i.xml:136]: This looks like a binding bug.

File: doc/classes/String.xml [hunk 8]
```diff
@@ -397,17 +405,6 @@
 				Returns [code]true[/code] if the string ends with the given string.
 			</description>
 		</method>
-		<method name="erase">
```
Comment (akien-mga) [inline doc/classes/String.xml:400]: Missing binding, or removed on purpose?

File: doc/classes/String.xml [hunk 9]
```diff
@@ -512,20 +512,6 @@
 				[/codeblock]
 			</description>
 		</method>
-		<method name="humanize_size">
```
Comment (akien-mga) [inline doc/classes/String.xml:515]: Missing binding, or removed on purpose?

File: doc/classes/Vector2.xml [hunk 10]
```diff
@@ -314,15 +314,6 @@
 				Returns this vector slid along a plane defined by the given normal.
 			</description>
 		</method>
-		<method name="snapped">
```
Comment (akien-mga) [inline doc/classes/Vector2.xml:317]: Missing binding, or removed on purpose?

File: doc/classes/String.xml [hunk 11]
```diff
@@ -485,10 +482,13 @@
...
 			<return type="int">
 			</return>
+			<argument index="0" name="with_prefix" type="bool" default="true">
+			</argument>
 			<description>
-				Converts a string containing a hexadecimal number into an integer. Hexadecimal strings are expected to be prefixed with "[code]0x[/code]" otherwise [code]0[/code] is returned.
+				Converts a string containing a hexadecimal number into a decimal integer. If [code]with_prefix[/code] is [code]true[/code], the hexadecimal string should start with the [code]0x[/code] prefix, otherwise [code]0[/code] is returned.
 				[codeblock]
 				print("0xff".hex_to_int()) # Print "255"
+				print("ab".hex_to_int())   # Print "171"
```
Comment (akien-mga) [inline doc/classes/String.xml:491]: Hadn't tested my own changes but I forgot the argument here:
```suggestion
				print("ab".hex_to_int(false)) # Print "171"
```

File: doc/classes/Basis.xml (hunk 1 above)
Comment (akien-mga) [reply doc/classes/Basis.xml:208]: Discussed it with @reduz, this was done on purpose as these are too complex to bind in the new system, and the multiplication operators can and should be used instead:
```
xform == mat * vec
xform_inv == vec * mat
```

Comment (akien-mga) [reply doc/classes/Basis.xml:208]: I also removed `Quat.xform` which had been kept.

File: doc/classes/Color.xml (hunk 2 above)
Comment (akien-mga) [reply doc/classes/Color.xml:198]: It's actually commented out:
```
// FIXME: Color is immutable, need to probably find a way to do this via constructor
//ADDFUNC4R(COLOR, COLOR, Color, from_hsv, FLOAT, "h", FLOAT, "s", FLOAT, "v", FLOAT, "a", varray(1.0));
```

File: doc/classes/Geometry2D.xml (hunk 3 above)
Comment (akien-mga) [reply doc/classes/Geometry2D.xml]: Fixed.

File: doc/classes/PackedByteArray.xml (hunk 4 above)
Comment (akien-mga) [reply doc/classes/PackedByteArray.xml:23]: It's not an intended change but it comes from the C++ API. We now have to make sure that the C++ API matches what we want to expose to the scripting API.

Here I'm not sure there's any C++ code that relies on those return values, so they could possibly be changed to `void` to preserve the previous scripting API.

File: doc/classes/PackedColorArray.xml (hunk 5 above)
Comment (akien-mga) [reply doc/classes/PackedColorArray.xml]: Binding had been missed, fixed.

File: doc/classes/Quat.xml (hunk 6 above)
Comment (akien-mga) [reply doc/classes/Quat.xml:154]: Commented out in `variant_call.cpp`:
```
// FIXME: Quat is atomic, this should be done via construcror
//ADDFUNC1(QUAT, NIL, Quat, set_euler, VECTOR3, "euler", varray());
//ADDFUNC2(QUAT, NIL, Quat, set_axis_angle, VECTOR3, "axis", FLOAT, "angle", varray());
```

File: doc/classes/Rect2i.xml (hunk 7 above)
Comment (akien-mga) [reply doc/classes/Rect2i.xml]: Fixed.

File: doc/classes/String.xml (hunk 8 above)
Comment (akien-mga) [reply doc/classes/String.xml:400]: Commented out in `variant_call.cpp`:
```
// FIXME: String needs to be immutable when binding
//bind_method(String, erase, sarray("position", "chars"), varray());
```

File: doc/classes/String.xml (hunk 9 above)
Comment (akien-mga) [reply doc/classes/String.xml:515]: Commented out in `variant_call.cpp`:
```
// FIXME: Static function, not sure how to bind
//bind_method(String, humanize_size, sarray("size"), varray());
```

File: doc/classes/Vector2.xml (hunk 10 above)
Comment (akien-mga) [reply doc/classes/Vector2.xml]: Was missed, fixed.

File: doc/classes/String.xml (hunk 11 above)
Comment (akien-mga) [reply doc/classes/String.xml]: Fixed.

**Comment by akien-mga** [comment]
Updated with classref changes from #43297 (+ added missing bindings).

**Review by reduz — APPROVED** [review]

**Comment by akien-mga** [comment]
I opened #43311 to keep track of the lost method bindings.

File: doc/classes/Quat.xml (hunk 6 above)
Comment (aaronfranke) [reply doc/classes/Quat.xml:154]: It seems that #30744 finally got resolved in core :)

**Pull request merged**